### PR TITLE
Move unboxed tuples to stable

### PR DESCRIPTION
**Pull request by ccasin**
This moves unboxed tuples to the "stable" maturity of the layouts extension, and adds very basic documentation.

### CODE REVIEW

**Review by goldfirere — COMMENTED** [review]
I have reviewed the docs and like them

File: testsuite/tests/typing-layouts-products/unboxed_tuples.ml [hunk 1]
```diff
@@ -3,19 +3,6 @@
  include stdlib_upstream_compatible;
  flambda2;
  {
-   ocamlc_byte_exit_status = "2";
-   setup-ocamlc.byte-build-env;
-   compiler_reference = "${test_source_directory}/unboxed_tuples_stable.compilers.reference";
```
Comment (goldfirere) [inline testsuite/tests/typing-layouts-products/unboxed_tuples.ml:8]: Should that output file also get deleted?

File: typing/jkind.ml [hunk 2]
```diff
@@ -793,11 +793,12 @@ module Const = struct
   let get_required_layouts_level (_context : History.annotation_context)
       (jkind : t) : Language_extension.maturity =
     match jkind.layout, jkind.nullability_upper_bound with
-    | (Base (Float64 | Float32 | Word | Bits32 | Bits64 | Vec128) | Any), _
+    | ( ( Base (Float64 | Float32 | Word | Bits32 | Bits64 | Vec128)
+        | Any | Product _ ),
```
Comment (goldfirere) [inline typing/jkind.ml:797]: Does this need to recur into `Product`?

**Review by goldfirere — APPROVED** [review]

File: typing/jkind.ml (hunk 2 above)
Comment (ccasin) [reply typing/jkind.ml]: Good catch - I've also added a test now.

**Pull request merged**